src/client/v2/algod/algod.ts

```diff
@@ -22,20 +22,12 @@ import Proof from './proof';
 
 export default class AlgodClient extends ServiceClient {
   constructor(
-    token = '',
+    token: string | AlgodTokenHeader,
```

Comment:
Here and in the indexer client, `token` should be able to accept a `CustomTokenHeader` too.